### PR TITLE
feat: Quickstart 3.0

### PR DESCRIPTION
**Pull request by jdjkelly**
## Why
* We are preparing to launch a new Quickstart guide and part of that release involves a new version of the source repository

## What
* Moves the example into one single NextJS project instead of having two applications
* New functionality for the Quickstart with API usage examples directly in app

----

> [!IMPORTANT]
> Consolidates Quickstart into a single Next.js project with new API functionalities, updated configurations, and enhanced documentation.
> 
>   - **Project Structure**:
>     - Consolidates into a single Next.js project, removing separate client and server directories.
>     - Deletes `client` and `server` directories.
>   - **API Endpoints**:
>     - Adds new API routes in `src/app/api` for `exchange`, `fhir/Coverage`, `fhir/ExplanationOfBenefit`, `fhir/Patient`, and `fhir/Patient/$everything`.
>     - Implements session management in `src/app/lib/session.ts`.
>   - **Configuration**:
>     - Adds `Dockerfile` for building and running the application.
>     - Updates `package.json` with new dependencies and scripts for Next.js.
>     - Configures ESLint and Tailwind CSS with `.eslintrc.json` and `tailwind.config.ts`.
>   - **UI Components**:
>     - Introduces UI components in `src/components/ui` for `Badge`, `Button`, `Card`, `Input`, `Select`, `Separator`, and `Table`.
>     - Implements API request components in `src/components/api-requests`.
>   - **Documentation**:
>     - Updates `README.md` to reflect new setup and usage instructions.
>     - Removes old `.env.template` files and updates environment variable handling.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=flexpa%2Fquickstart&utm_source=github&utm_medium=referral)<sup> for c6e4ef2ad59de73087767a23d917c9aa96fe1ed8. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

**Review by larry-flexpa — CHANGES_REQUESTED** [review]
Is this missing the env.example?

File: src/app/dashboard/page.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+  const decoded = decodeJwt(token?.accessToken as string);
+
+  return (
+    <div className="min-h-screen flex items-center justify-center bg-white">
+      <div className="max-w-2xl mx-auto px-4">
+        <h1 className="text-4xl font-bold mb-2">Flexpa Quickstart</h1>
+        <p className="text-xl mb-6">Patient Authorization</p>
+        <p className="text-gray-600 mb-8">
+          Congrats! By linking a health insurance plan, you have created a <a href="https://docs.flexpa.com/reference/patient-authorization" className="text-black font-semibold underline">Patient Authorization</a>.
```
Comment (larry-flexpa) [inline src/app/dashboard/page.tsx:21]: Our docs don't have a reference here. Also our docs are at `https://www.flexpa.com/docs`

**Review by larry-flexpa — APPROVED** [review]
Only problem is the patient authorization link doesn't go anywhere, but otherwise looks great! Excited to use this.

File: src/app/dashboard/page.tsx (hunk 1 above)
Comment (jdjkelly) [reply src/app/dashboard/page.tsx:21]: Hallucination...

Comment (jdjkelly) [inline src/app/dashboard/page.tsx:21]:
```suggestion
          Congrats! By linking a health insurance plan, you have created a Patient Authorization.
```

**Pull request merged**